[ixia/Keysight] Adding IxANVL automated deployment using PTF32 topology.

aganguly-keysight opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/Azure/SONiC/blob/gh-pages/CONTRIBUTING.md

Please provide following information to help code review process a bit easier:
-->
### Description of PR
Addition of IxANVL PTF Container automated deployment using PTF32 topology.

Summary:
Fixes # (issue)

### Type of change

<!--
- Fill x for your type of change.
- e.g.
- [x] Bug fix
-->

- [ ] Bug fix
- [ x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Approach
#### What is the motivation for this PR?

#### How did you do it?
Modified ansible/roles/vm_set/tasks/add-topo.yml, ansible/roles/vm_set/tasks/remove-topo.yml, ansible/roles/vm_set/tasks/main.yml to create and bind IxANVL container in a PTF32 topology with SONIC-VS DUT.
Added a new configuration - "ixanvl-vs-conf" in testbed.csv and testbed.yaml for this.

Found one issue during PTF32 topology deployment as the bind was throwing errors as there was no VMs needed for this topology and hence was not defined in topology ymls. Fixed that error by making "vm_names" to be an optional argument in 
ansible/roles/vm_set/library/vm_topology.py main function. Made subsequent changes in that file for this purpose.

#### How did you verify/test it?
By running testbed-cli.sh script for add-topo and remove-topo for ixanvl-vs-conf configuration.
#### Any platform specific information?

#### Supported testbed topology if it's a new test case?

### Documentation 
A new README added in testbed section.
Readme file: README.testbed.IxANVL.VsSetup.md

yxieca commented on ansible/roles/vm_set/library/vm_topology.py:
```diff
@@ -923,7 +922,7 @@ def main():
             vm_set_name=dict(required=False, type='str'),
             topo=dict(required=False, type='dict'),
-            vm_names=dict(required=True, type='list'),
+            vm_names=dict(required=False, type='list'),
```
I think @neethajohn asked for a similar change in another PR. please don't allow skip this option but allow provide empty list.

yxieca commented on ansible/roles/vm_set/library/vm_topology.py:
```diff
@@ -941,10 +940,13 @@ def main():
+    if 'vm_names' not in module.params:
+        vm_names = []
+    else:
+        vm_names = module.params['vm_names']
```
As above.

aganguly-keysight replied:
Hi @yxieca 

I have updated vm_topology.py as per the review comments. 
Please share your feedback on the latest changes.

aganguly-keysight commented:
Hello @yxieca 
I have updated the PR with changes for running bgp conformance tests. Updated conftest.py and added test_bgp_conformance.py . Please share your review comments.
